Issue #83) Added Debug Mode

jldistefano opened the pull request:
Please test the `-d` flag to see that log messages appear in console, and `-l` should generate a file and directory in `project_aurum/client/logs/logs.txt`. `-ltestfile.txt` should create a file and directory `project_aurum/client/logs/testfile.txt`.

kstola2 commented on client/src/main/main.go:
```diff
@@ -4,42 +4,73 @@ import (
+	//"strings"
 
-	client "../client"
+	client "project_aurum/client/src/client"
```
This import path is causing my file to fail to compile. Is the old version doing the same for you?

jldistefano replied:
It doesn't do it for me, but that's likely because of where my project is installed. That's just an artifact from the branch talking about using absolute import paths rather than relative. But I'll change it back.

jldistefano replied:
The branch is now updated

kstola2 commented:
So debug works great, but log isn't working for some reason. It just quits if I dont give `-l` and argument, and from what I'm assuming it should generate the `logs.txt` file if I do that right?

jldistefano commented:
When you say it just quits do you mean if you run the command `./main` without a -l it just crashes the program? And the command `./main l` doesn't create a file for you in `project_aurum/client/logs` called `logs.txt`?

kstola2 commented:
![image](https://user-images.githubusercontent.com/31855040/54165502-fb7d7b80-442e-11e9-991c-37cb238f4dd7.png)

Also no `logs` directory is created in any of these cases.

jldistefano commented:
For a start, if you want to specify a filename using the short flag you have to do `-lfilename`, the long ones use `--logfile=filename`. I've updated the branch with a possible fix, which uses the os.MkdirAll command to make the directory, and uses a relative filepath instead of relying on the GOPATH variable. I talk a lot about the relative pathing stuff, but if this fixes it then it is no less broken then the rest of the functioning code we've got right now. I also removed some accidental debugging print statements. Let me know if that works.

kstola2 commented on client/src/main/main.go:
```diff
@@ -4,42 +4,75 @@ import (
+	//err := error(nil) // This is used for offline testing
 	if err != nil {
-		log.Fatalln("Connectivity check failed.")
+		logger.Fatalln(err)
```
Everything looks good, but why was this error statement changed?

jldistefano replied:
I thought it'd be more descriptive to give the actual error instead of just saying that it failed. The error could give more information than I am capable of.
